packages/core/src/BlockNoteEditor.ts

```diff
@@ -216,28 +216,36 @@ export class BlockNoteEditor<BSchema extends BlockSchema = DefaultBlockSchema> {
     const tiptapOptions: EditorOptions = {
       ...blockNoteTipTapOptions,
       ...newOptions._tiptapOptions,
-      onCreate: () => {
-        newOptions.onEditorReady?.(this);
-        this.ready = true;
-      },
       onBeforeCreate(editor) {
         if (!initialContent) {
           // when using collaboration
           return;
         }
-        // we have to set the initial content here, because now we can use the editor schema
-        // which has been created at this point
-        const schema = editor.editor.schema;
-        const ic = initialContent.map((block) => blockToNode(block, schema));
 
+        // We always set the initial content to a single paragraph block. This
+        // allows us to easily replace it with the actual initial content once
+        // the TipTap editor is initialized.
+        const schema = editor.editor.schema;
         const root = schema.node(
           "doc",
           undefined,
-          schema.node("blockGroup", undefined, ic)
+          schema.node("blockGroup", undefined, [
+            blockToNode({ id: "initialBlock", type: "paragraph" }, schema),
+          ])
         );
-        // override the initialcontent
         editor.editor.options.content = root.toJSON();
       },
+      onCreate: () => {
+        newOptions.onEditorReady?.(this);
```

Comment:
I think we should move this to after replaceBlocks. Because the user will expect the blocks to have the content of `initialContent` when `onEditorReady` is called